osquery/tables/system/windows/prefetch.cpp

```diff
@@ -0,0 +1,408 @@
+  std::string directories;
+  size_t directories_count;
+  std::string volume_serial;
+  std::string volume_creation;
+};
+
+#pragma pack(push, 1)
+typedef struct _PREFETCH_COMPRESSED_HEADER {
+  DWORD Signature;
+  DWORD TotalUncompressedSize;
+  BYTE CompressedData[1]; // arbitrary size
+} PREFETCH_COMPRESSED_HEADER, *PPREFETCH_COMPRESSED_HEADER;
+
+typedef struct _PREFETCH_FILE_HEADER {
+  DWORD Version;
+  DWORD Signature;
+  DWORD Reserved1;
+  DWORD FileSize;
+  WCHAR FileName[30];
+  DWORD Hash;
+  DWORD Reserved2;
+} PREFETCH_FILE_HEADER, *PPREFETCH_FILE_HEADER;
+
+typedef struct _PREFETCH_FILE_INFORMATION {
+  DWORD FileMetricsArrayOffset;
+  DWORD NumberOfMetricEntries;
+  DWORD TraceChainsArrayOffset;
+  DWORD NumberOfTraceChains;
+  DWORD FileNameStringsOffset;
+  DWORD FileNameStringsSize;
+  DWORD VolumeInformationOffset;
+  DWORD NumberOfVolumes;
+  DWORD VolumesInformationSize;
+  union {
+    struct {
+      DWORD Reserved1[2];
+      FILETIME LastRunTime;
+      DWORD Reserved2[4];
+      DWORD RunCount;
+    } v23;
+    struct {
+      DWORD Reserved1[2];
+      FILETIME LastRunTime;
+      FILETIME OtherRunTimes[7];
+      DWORD Reserved2[4];
+      DWORD RunCount;
+    } v26;
+    struct {
+      DWORD Reserved1[2];
+      FILETIME LastRunTime;
+      FILETIME OtherRunTimes[7];
+      DWORD Reserved2[2];
+      DWORD RunCount;
+    } v30v2;
+  } ext;
+} PREFETCH_FILE_INFORMATION, *PPREFETCH_FILE_INFORMATION;
+
+typedef struct _PREFETCH_VOLUME_INFORMATION {
+  DWORD VolumePathOffset;
+  DWORD VolumeDevicePathNumberOfCharacters;
+  FILETIME VolumeCreationTime;
+  DWORD VolumeSerialNumber;
+  DWORD FileReferencesOffset;
+  DWORD FileReferencesDataSize;
+  DWORD DirectoryStringsOffset;
+  DWORD NumberOfDirectoryStrings;
+  DWORD Reserved1;
+} PREFETCH_VOLUME_INFORMATION, *PPREFETCH_VOLUME_INFORMATION;
+
+typedef struct _DIRECTORY_STRING {
+  USHORT Size;
+  WCHAR Directory[1]; // arbitrary size
+} DIRECTORY_STRING, *PDIRECTORY_STRING;
+#pragma pack(pop)
+
+} // namespace
+
+PrefetchHeader parseHeader(const PREFETCH_FILE_HEADER* header) {
+  PrefetchHeader result;
+  if (header->FileName[(sizeof(header->FileName) / sizeof(WCHAR)) - 1] ==
+      '\0') {
+    result.filename = wstringToString(header->FileName);
+  }
+  result.prefetch_hash = (boost::format("%x") % header->Hash).str();
+  result.file_size = header->FileSize;
+  return result;
+}
+
+PrefetchFileInfo parseFileInfo(
+    const std::vector<UCHAR>& data,
+    const PREFETCH_FILE_INFORMATION* prefetch_file_info,
+    DWORD version) {
+  PrefetchFileInfo result;
+
+  FILETIME last_run_time;
+  std::vector<std::string> run_times;
+
+  switch (version) {
+  case kPrefetchVersionWindows10:
+    last_run_time = prefetch_file_info->ext.v30v2.LastRunTime;
+    result.run_count = prefetch_file_info->ext.v30v2.RunCount;
+    for (const auto& entry : prefetch_file_info->ext.v30v2.OtherRunTimes) {
+      LONGLONG time = filetimeToUnixtime(entry);
+      if (time != -11644473600) {
```

Comment:
What is this value? I see it in the code base a few times?

Reply:
its the value returned if FILETIME is 0 (Jan 01 1601 00:00:00)